crates/tests/implement_core/src/impl_on_outer.rs

```diff
@@ -0,0 +1,63 @@
+use windows_core::*;
+
+#[interface("cccccccc-0000-0000-0000-000000000001")]
+unsafe trait IFoo: IUnknown {
+    fn hello(&self);
+}
+
+#[interface("cccccccc-0000-0000-0000-000000000002")]
+unsafe trait IFoo2: IFoo {
+    fn hello(&self);
+}
+
+#[interface("cccccccc-0000-0000-0000-000000000003")]
+unsafe trait IFoo3: IFoo2 {
+    fn hello(&self);
+}
+
+#[interface("cccccccc-0000-0000-0000-000000000004")]
+unsafe trait IBar: IUnknown {
+    fn goodbye(&self);
+}
+
+// This tests that we can compile a COM object that has some COM interfaces implemented on the
+// outer object and some on the inner object.
+#[implement(ImplLocation = Outer, IFoo3, ImplLocation = Inner, IBar)]
+struct MyApp {}
+
+impl IFoo_Impl for MyApp_Impl {
+    unsafe fn hello(&self) {
```

Comment:
Would like to see these tests verify that this is the correct `self` type

Reply:
This test does verify that.  If you get it wrong, it won't compile.